processor/src/test/java/org/mapstruct/ap/internal/util/StringsTest.java

```diff
@@ -0,0 +1,99 @@
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.internal.util;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import org.junit.Test;
+
+/**
+ * @author Filip Hrisafov
+ */
+public class StringsTest {
+    @Test
+    public void testCapitalize() throws Exception {
+
+        assertThat( Strings.capitalize( null ) ).isNull();
+        assertThat( Strings.capitalize( "c" ) ).isEqualTo( "C" );
+        assertThat( Strings.capitalize( "capitalize" ) ).isEqualTo( "Capitalize" );
+        assertThat( Strings.capitalize( "AlreadyCapitalized" ) ).isEqualTo( "AlreadyCapitalized" );
+        assertThat( Strings.capitalize( "notCapitalized" ) ).isEqualTo( "NotCapitalized" );
+
+    }
+
+    @Test
+    public void testDecapitalize() throws Exception {
+        assertThat( Strings.decapitalize( null ) ).isNull();
+        assertThat( Strings.decapitalize( "c" ) ).isEqualTo( "c" );
+        assertThat( Strings.decapitalize( "capitalize" ) ).isEqualTo( "capitalize" );
+        assertThat( Strings.decapitalize( "AlreadyCapitalized" ) ).isEqualTo( "alreadyCapitalized" );
+        assertThat( Strings.decapitalize( "notCapitalized" ) ).isEqualTo( "notCapitalized" );
+    }
+
+    @Test
+    public void testJoin() throws Exception {
+        assertThat( Strings.join( new ArrayList<String>(), "-" ) ).isEqualTo( "" );
+        assertThat( Strings.join( Arrays.asList( "Hello", "World" ), "-" ) ).isEqualTo( "Hello-World" );
+        assertThat( Strings.join( Arrays.asList( "Hello" ), "-" ) ).isEqualTo( "Hello" );
+    }
+
+    @Test
+    public void testJoinAndCamelize() throws Exception {
+        assertThat( Strings.joinAndCamelize( new ArrayList<String>() ) ).isEqualTo( "" );
+        assertThat( Strings.joinAndCamelize( Arrays.asList( "Hello", "World" ) ) ).isEqualTo( "HelloWorld" );
+        assertThat( Strings.joinAndCamelize( Arrays.asList( "Hello", "world" ) ) ).isEqualTo( "HelloWorld" );
+        assertThat( Strings.joinAndCamelize( Arrays.asList( "hello", "world" ) ) ).isEqualTo( "helloWorld" );
+    }
+
+    @Test
+    public void testIsEmpty() throws Exception {
+        assertThat( Strings.isEmpty( null ) ).isTrue();
+        assertThat( Strings.isEmpty( "" ) ).isTrue();
+        assertThat( Strings.isEmpty( "not empty" ) ).isFalse();
+
```

Comment:
Well this actually is not empty in our implementation. We do a `null` check and then `String#isEmpty()`